Vanilla Circuit Connector Sprites

JDOGG88 opened the pull request:
![image](https://user-images.githubusercontent.com/6249575/54778367-7179a380-4be2-11e9-9bcd-4a95fdc6c9b7.png)

dgw commented on prototypes/entity.lua:
```diff
@@ -19,6 +19,43 @@ if settings.startup["Warehousing-sixteen-mode"].value then
+end
+function shiftY(modifier_y, y)
+	return modifier_y + y
+end
```
These functions are redundant. They take the same arguments and return the same result, just with different names.

Honestly, it would be clearer to just embed the arithmetic below, instead of obfuscating it with these. I think it would also be clearer (and cleaner) to put these shift vectors in the declared prototypes.

JDOGG88 replied:
I realized these were redundant later on but I ran out of time to continue working on it.

The reason I wanted the shift vectors outside calling a function was so if we needed to move it all we had to do was change 2 values instead of each individual sprite. So it was an attempt to future proof the new models using these.

JDOGG88 replied:
I'm slow lol, I understand what you mean and it will still work the same as I have it now haha. I still want to screw around with it to see if there's any way I can make this cleaner with a function to generate the Sprite callbacks. Or possibly put it in a new file and call that file into the entity.lua file. Not sure yet what will be easiest and cleanest to function as efficiently as possible. Another puzzle for me to solve 😊😊

JDOGG88 commented:
Unfortunately how I have it set up atm yes it would pass through to the logistic models, I just didn't have enough time to set the logistic ones up properly to use their antennas.

As far as looking up the other code I'm on my phone and probably won't be able to work on it much if at all this weekend. But I'll definitely look into it when I can.

JDOGG88 commented:
![image](https://user-images.githubusercontent.com/6249575/54956263-90976e80-4f1d-11e9-9535-1093f5a2eaa0.png)

JDOGG88 commented:
I was able to separate the logistics wiring out as it's own connection points.

Also I added a functions for the connector and wires so it eliminated ~60 duplicate lines of code and it will be easy to just pop the variables into the small and tiny versions too.

JDOGG88 commented:
I did a lot of code manipulation so I'm not sure if you would technically want this in this PR but I feel like what I did will be necessary when the small and tiny versions are implemented.

I changed the base storehouse and warehouse creation to a function so it eliminated 60 more lines of duplicate code.  It will also make it so we would only need to add 16 lines for the small and tiny versions instead for the 200+ lines how I have it now.

I also added some if then statements to clean up the logistic version creation so as to make the small and tiny versions easier to add in.

dgw commented:
This stalled out for so long, someone came along and made us new sprites. See #88.
